From cc08843519a0f2da631ec57603bbb1cc1724d9cb Mon Sep 17 00:00:00 2001
From: wyuan <wyuan@codeaurora.org>
Date: Wed, 27 May 2015 16:35:05 +0800
Subject: [PATCH 104/296] Remove freezing window to fix UI freezing issue

In some corner case, some window is at exiting state, but can't be
removed. because the binder transact failed, and the state of window can't change
so that will still resizing window,This will result the windownmanger fall into freezing state

The fix will stop resizing this window when encounter this case
---
 services/core/java/com/android/server/wm/WindowState.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index a3c378ddb84..088fbc77c79 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -3188,6 +3188,11 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
             mWinAnimator.mSurfaceResized = false;
             mReportOrientationChanged = false;
         } catch (RemoteException e) {
+            mOverscanInsetsChanged = false;
+            mContentInsetsChanged = false;
+            mVisibleInsetsChanged = false;
+            mStableInsetsChanged = false;
+            mWinAnimator.mSurfaceResized = false;
             setOrientationChanging(false);
             mLastFreezeDuration = (int)(SystemClock.elapsedRealtime()
                     - mService.mDisplayFreezeTime);
-- 
2.11.0

